File: sorting/insertion_sort/insertionsort.ts
function InsertionSort(unsortedArray: number[]): number[] {
    let n: number = unsortedArray.length;

    for (let i = 0; i < n - 1; i ++) {
        let min: number = i;
        
        for (let j = i + 1; j < n; j ++) {
            if (unsortedArray[j] < unsortedArray[min]) {
                min = j;
            }
        }

        let temp: number = unsortedArray[i];
        unsortedArray[i] = unsortedArray[min];
        unsortedArray[min] = temp;
    }

    return unsortedArray;
}

File: sorting/insertion_sort/InsertionSort.java
public class Sorting {
    public static int[] InsertionSort(int[] unsortedArray) {
        int n = unsortedArray.length;
        int i = 0;
        int j, min, temp;

        while (i < n - 1) {
            min = i;
            j = i;
            
            while (j < n) {
                if (unsortedArray[j] < unsortedArray[min]) {
                    min = j;
                }
            }

            temp = unsortedArray[min];
            unsortedArray[min] = unsortedArray[i];
            unsortedArray[i] = temp;
        }

        return unsortedArray;
    }
}

class Test {
    
}